app/controllers/CheckYourAnswersController.scala

```diff
@@ -128,42 +127,47 @@ class CheckYourAnswersController @Inject()(checkYourAnswersPage: CheckYourAnswer
-        val confirmationEndDate = userRequest.answers.getAnswer[LocalDate](SessionKeys.endDateOfPeriod).get.toString
-        val confirmationObligation = userRequest.answers.getAnswer[Boolean](SessionKeys.isObligationAppeal).contains(true).toString
-        val confirmationMultipleAppeals = userRequest.answers.getAnswer[String](SessionKeys.doYouWantToAppealBothPenalties).getOrElse("no")
-        val confirmationIsAgent = userRequest.isAgent.toString
-        val penaltiesHasSeenConfirmationPage = "true"
-        uploadJourneyRepository.removeUploadsForJourney(userRequest.session.get(SessionKeys.journeyId).get).map {
-          _ => Redirect(controllers.routes.CheckYourAnswersController.onPageLoadForConfirmation()).addingToSession(
-            List(
-              (SessionKeys.confirmationAppealType, confirmationAppealType),
-              (SessionKeys.confirmationStartDate, confirmationStartDate),
-              (SessionKeys.confirmationEndDate, confirmationEndDate),
-              (SessionKeys.confirmationObligation, confirmationObligation),
-              (SessionKeys.confirmationMultipleAppeals, confirmationMultipleAppeals),
-              (SessionKeys.penaltiesHasSeenConfirmationPage, penaltiesHasSeenConfirmationPage),
-              (SessionKeys.confirmationIsAgent, confirmationIsAgent)): _*
-          )
-        }
+        val previouslySubmittedJourneyId = userRequest.answers.journeyId
+        Future(Redirect(controllers.routes.CheckYourAnswersController.onPageLoadForConfirmation()).addingToSession(
+          SessionKeys.previouslySubmittedJourneyId -> previouslySubmittedJourneyId,
+          SessionKeys.penaltiesHasSeenConfirmationPage -> "true"
+        ))
       }
     ))
   }
 
-  def onPageLoadForConfirmation(): Action[AnyContent] = authorise {
-    implicit userRequest => {
-      val appealType = PenaltyTypeEnum.withName(userRequest.session.get(SessionKeys.confirmationAppealType).get)
-      val bothPenalties: String = userRequest.session.get(SessionKeys.confirmationMultipleAppeals).get
-      val startDate: LocalDate = LocalDate.parse(userRequest.session.get(SessionKeys.confirmationStartDate).get)
-      val endDate: LocalDate = LocalDate.parse(userRequest.session.get(SessionKeys.confirmationEndDate).get)
-      val (readablePeriodStart, readablePeriodEnd) = (dateToString(startDate), dateToString(endDate))
-      val isObligationAppeal: Boolean = userRequest.session.get(SessionKeys.confirmationObligation).get.toBoolean
-      val showDigitalCommsMessage: Boolean = isEnabled(ShowDigitalCommsMessage)
-      val isAgent: Boolean = userRequest.session.get(SessionKeys.confirmationIsAgent).get.toBoolean
-      val vrn= userRequest.vrn
-      Ok(appealConfirmationPage(readablePeriodStart, readablePeriodEnd, isObligationAppeal, showDigitalCommsMessage, appealType, bothPenalties, isAgent, vrn))
-        .removingFromSession(SessionKeys.allKeys: _*)
+  def onPageLoadForConfirmation(): Action[AnyContent] = authorise.async {
```

Comment:
Maybe not one for now but possibly should move this to its own controller, or a post submission controller with appeals details page

Reply:
Good idea, something like AppealConfirmationController? Will keep me busy :)